### PR TITLE
Fix downcase issue with NTLMHash imports

### PR DESCRIPTION
**Pull request by zgoldman-r7**

### CODE REVIEW

**Comment by jmartin-r7** [comment]
This looks like it could impact at least [two other private types](https://github.com/rapid7/metasploit-credential/search?q=normalize_data). 

`PostgresMD5` and `KrbEncKey`

**Comment by zgoldman-r7** [comment]
@jmartin-r7 looks like `KrbEncKey` might not be a concern as it's not listed in `lib/metasploit/credential/importer/base.rb` under the allowed importer types:
```
  LONG_FORM_ALLOWED_PRIVATE_TYPE_NAMES = [
      Metasploit::Credential::NonreplayableHash,
      Metasploit::Credential::NTLMHash,
      Metasploit::Credential::Password,
      Metasploit::Credential::PostgresMD5,
      Metasploit::Credential::SSHKey].map(&:name)


  # Whitelist of the {Metasploit::Credential::Private} subclass names allowed
  # in short-form CSV imports.
  SHORT_FORM_ALLOWED_PRIVATE_TYPE_NAMES = [
      Metasploit::Credential::NonreplayableHash,
      Metasploit::Credential::NTLMHash,
      Metasploit::Credential::Password,
      Metasploit::Credential::PostgresMD5].map(&:name)
```

**Comment by jmartin-r7** [comment]
I suspect that is an over site in the recent addition of `KrbEncKey` either way it may be reasonable to check the class for normalization needs and hand off any normalization process needed to make change here more resilient to addition of future types.

File: lib/metasploit/credential/importer/core.rb
Comment (jmartin-r7) [inline lib/metasploit/credential/importer/core.rb:1]: Please remove commented out code and stray whitespace.

File: spec/lib/metasploit/credential/creation_spec.rb [hunk 1]
```diff
@@ -178,6 +178,37 @@
...
+        private_type: :postgres_md5
+      }}
+      it 'allows for the recreation of core with case insensitive private credentials set to different case' do
+        expect{ test_object.create_credential(credential_data) }.to change{ Metasploit::Credential::PostgresMD5.count }.by(1)
+        expect{ Metasploit::Credential::Core.first.destroy }.to change{ Metasploit::Credential::Core.count }.by(-1)
+        expect( Metasploit::Credential::PostgresMD5.count ).to eq(1)
+        expect( Metasploit::Credential::Core.count ).to eq(0)
+        expect{ test_object.create_credential(credential_data_upcase) }.to change{ Metasploit::Credential::Core.count }.by(1)
+        expect( Metasploit::Credential::PostgresMD5.count ).to eq(1)
+      end
```
Comment (jmartin-r7) [inline spec/lib/metasploit/credential/creation_spec.rb:210]: This matches with current expectations, should we expand `Core.destroy` to remove `public` and `private` objects that no longer have a core attached to them?

The scenario I am thinking of would be:
* add a cred with `user1` and `private1`
* add a second cred with `user2` and `private1`
* expect 2 cores,  2 publics, 1 private in db
* remove core for `user1`
* remove core for `user2`
* expect all cores, publics, private have been removed from db

And:
* add a cred with `user1` and `private1`
* add a second cred with `user2` and `private1`
* add a third cred with `user3` and `private1`
* expect 3 cores,  3 publics, 1 private in db
* remove core for `user1`
* remove core for `user2`
* expect 1 core, 1 public, 1 private are left in the db

Given the current patterns in place I can see scenarios where a core has both a `PasswordHash` and `Password` added by a cracking run. In this case when the core is deleted we would still have both privates values in the database still with no way know that the ``PasswordHash` and `Password` relate.

This could be deferred for later instead of part of this PR, just thought it is worth considering now.

Comment (jmartin-r7) [reply spec/lib/metasploit/credential/creation_spec.rb:210]: After some investigation, decision is to accept the adjustment here as is and opt to address `orphaned` data in a separate PR.
